pkg/segment/aggregations/generateevents.go

```diff
@@ -86,3 +99,112 @@ func PerformGenTimes(aggs *structs.QueryAggregators) error {
 
 	return nil
 }
+
+func checkCSVFormat(filename string) bool {
+	return strings.HasSuffix(filename, ".csv") || strings.HasSuffix(filename, ".csv.gz")
+}
+
+// Ensure record and column lengths are valid before calling this function.
+func createRecord(columnNames []string, record []string) map[string]interface{} {
+	recordMap := make(map[string]interface{})
+	for i, col := range columnNames {
+		recordMap[col] = record[i]
+	}
+	return recordMap
+}
+
+func PerformInputLookup(aggs *structs.QueryAggregators) error {
+	var reader *csv.Reader
```

Comment:
nit: `reader` isn't used for a while, so you might want to declare it just before it's actually set up (around line 139 I think).

However, you can keep what you have if you prefer this way

Reply:
addressed in the latest commit.